### PR TITLE
fix: [FFM-7866]: replace Feature Flag CFNG_ENABLED with proper licensing

### PR DESCRIPTION
**Pull request by Kira-Jennings**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by Kira-Jennings** [comment]
retrigger titlecheck

**Comment by Kira-Jennings** [comment]
retrigger titlecheck

File: src/modules/10-common/navigation/MainNav/__tests__/MainNav.test.tsx [hunk 1]
```diff
@@ -104,12 +104,11 @@ describe('main nav tests', () => {
     expect(queryByText('featureFlagsText')).not.toBeNull()
   })
 
-  test('when ng dashbpard is enabled', () => {
+  test('when ng dashb0ard is enabled', () => {
```
Comment (knagurski) [inline src/modules/10-common/navigation/MainNav/__tests__/MainNav.test.tsx:107]: One typo replaced by another typo lol
```suggestion
  test('when ng dashboard is enabled', () => {
```

**Review by knagurski — DISMISSED** [review]
Dang, that was a lot of files. Great work and good idea switching from strings to the enum for the comparisons. One possible thing to think about is how we can maintain the previous snapshots... when removing the `CFNG_ENABLED` flag from the `TestWrapper`, could we add the license to the `defaultLicenseStoreValues` so that the tests still get the same values?

**Comment by Kira-Jennings** [comment]
> could we add the license to the defaultLicenseStoreValues so that the tests still get the same values?

@knagurski  I did this originally and then realised that most of the tests didn't need or rely on our module at all, and that the tests pass without the `defaultLicenseStoreValues`, so I removed them as they seem redundant

File: src/modules/10-common/navigation/MainNav/__tests__/MainNav.test.tsx (hunk 1 above)
Comment (Kira-Jennings) [reply src/modules/10-common/navigation/MainNav/__tests__/MainNav.test.tsx:107]: OOPS

**Review by crystal-fong — APPROVED** [review]

**Review by Harness-JKhattak — DISMISSED** [review]

**Review by sonaligoyal-harness — APPROVED** [review]

**Review by tan-nhu — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

**Review by YounixIO — APPROVED** [review]

**Review by Harness-JKhattak — APPROVED** [review]

**Comment by Kira-Jennings** [comment]
@harness/ui-platform

**Comment by Kira-Jennings** [comment]
@harness/ui-plg

**Review by dhruvUpadhyay07 — APPROVED** [review]

**Review by sunnykesh-harness — APPROVED** [review]

**Pull request merged**